src/main/java/org/example/auctions/services/Auction.java

```diff
@@ -0,0 +1,106 @@
+package org.example.auctions.services;
+
+import java.util.*;
+
+public class Auction {
+    private final String id;
+    private String subject;
+    private type type;
+    private Integer price;
+    private duration duration;
+    private final String ownerName;
+    private final ArrayList<Bid> bids;
+    private String winnerBidId;
+    private status status;
+
+
+    public Auction(String subject, type type, Integer price, duration duration, String ownerName) {
```

Comment:
"type type" - is very confusing.
The names of the classes, interfaces, enums in Java should start with uppercase.
And even if you rename type into Type it is still confusing. Type is too general name, says nothing what you mean.
Better naming will be AuctionType, for example.

Reply:
The same application to duration duration.
Enum name should start with uppercase character.
Duration. Better naming is AuctionDuration.